client/header/transient-notices/index.js

```diff
@@ -0,0 +1,48 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { Component } from '@wordpress/element';
+import { compose } from '@wordpress/compose';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import TransientNotice from './transient-notice';
+import withSelect from 'wc-api/with-select';
+
+class TransientNotices extends Component {
+	render() {
+		const { className, notices } = this.props;
+		const classes = classnames( 'woocommerce-transient-notices', className );
+
+		return (
+			<section className={ classes }>
+				{ notices && notices.map( notice => <TransientNotice key={ notice.id } { ...notice } /> ) }
```

Comment:
minor but should we just use the option second iterator argument from Array.map to populate the key here since theoretically we don't have any logic in place to prevent duplicate key.ids?

Reply:
I think that’s probably a good idea. My original thought here was that we could pass in IDs to override existing messages. However since the use case for that may be slim, we can probably remove it.